### PR TITLE
Demo T5 model on sentiment classification and translation

### PR DESCRIPTION
**Pull request by pmabbo13**
# Description

Update T5 tutorial to include demonstrations of how to use model for sentiment classification on IMDB dataset and English to German translation on Multi30k dataset.

# Process

The IMDB dataset was loaded and processed so that the labels `neg` and `pos` were changed to `negative` and `positive` to align with the labels that would be generated by the model. The input texts were also appended with the prefix `sst2 sentence` to indicate that the task is sentiment classification (this is the prefix which T5 was trained on for this task).

The Multi30k datasets was loaded and processed so that the english text was pre-pended with the prefix `translate English to German` to indicate the task to the model.

The same logic used to generate the summarization sequences from CNNDM text was also used to generate the outputs for the IMDB and Multi30k texts. Note that for sentiment translation we use a `beam_size=1` since we are looking for the most probable sequence label, which should always be one word.

Also, given that the tutorial will now demo the T5 model on 3 tasks, I've changed the title to "T5-Base Model for Summarization, Sentiment Classification, and Translation", and the file name from `cnndm_summarization.py` to `t5_demo.py`.

# Testing

Run `BUILD_GALLERY=1 make 'SPHINXOPTS=-W' html` in `docs` and review rendered document in `docs/build/html/tutorials/t5_demo.html`

### CODE REVIEW

File: examples/tutorials/t5_demo.py [hunk 1]
```diff
@@ -374,3 +425,174 @@ def generate(encoder_tokens: Tensor, eos_idx: int, model: T5Model, beam_size: in
...
+#    target: negative
+#
+
+
+#######################################################################
+# Generate Translations
+# ---------------------
+#
+# Similarly, we can now use the model to generate sentiment classification on the first batch of reviews from the IMDB test set
+# using a beam size of 4.
```
Comment (parmeet) [inline examples/tutorials/t5_demo.py:530]: could you please update the text to reflect the translation task?

**Review by parmeet — APPROVED** [review]
Thanks @pmabbo13! Overall looks great :).

File: examples/tutorials/t5_demo.py [hunk 2]
```diff
@@ -374,3 +425,174 @@ def generate(encoder_tokens: Tensor, eos_idx: int, model: T5Model, beam_size: in
...
+# Generate Sentiment Classifications
+# ----------------------------------
+#
+# Similarly, we can now use the model to generate sentiment classifications on the first batch of reviews from the IMDB test set
+# using a beam size of 1.
+#
+
+batch = next(iter(imdb_dataloader))
+input_text = batch["text"]
+model_input = transform(input_text)
```
Comment (Nayef211) [inline examples/tutorials/t5_demo.py:441]: nit: can we move this line to be just above `model_output`? Same with the other tasks

File: examples/tutorials/t5_demo.py [hunk 3]
```diff
@@ -374,3 +425,174 @@ def generate(encoder_tokens: Tensor, eos_idx: int, model: T5Model, beam_size: in
...
+target = batch["label"]
+beam_size = 1
+
+model_output = generate(model=model, encoder_tokens=model_input, eos_idx=eos_idx, beam_size=beam_size)
+output_text = transform.decode(model_output.tolist())
+
+for i in range(batch_size):
+
```
Comment (Nayef211) [inline examples/tutorials/t5_demo.py:449]: nit: remove newline

File: examples/tutorials/t5_demo.py [hunk 4]
```diff
@@ -374,3 +425,174 @@ def generate(encoder_tokens: Tensor, eos_idx: int, model: T5Model, beam_size: in
...
+input_text = batch["english"]
+model_input = transform(input_text)
+target = batch["german"]
+beam_size = 4
+
+model_output = generate(model=model, encoder_tokens=model_input, eos_idx=eos_idx, beam_size=beam_size)
+output_text = transform.decode(model_output.tolist())
+
+for i in range(batch_size):
+
```
Comment (Nayef211) [inline examples/tutorials/t5_demo.py:543]: nit: remove new line

**Review by Nayef211 — APPROVED** [review]
LGTM as well. Can we just check why the sentiment output looks different from the summarization and translation output in the [generated doc](https://output.circle-artifacts.com/output/job/1ddeaffa-e3da-4759-a56f-29374921c445/artifacts/0/docs/tutorials/t5_demo.html#sentiment-output)

**Pull request merged**